src/Components/SignatureModal/SignatureModal.tsx

```diff
@@ -81,27 +81,20 @@ class SignatureModal extends React.PureComponent<Props> {
       </TransparentButton>
     );
   };
-  private renderNativeModal = () => (
-    <Modal
+  private renderModal = () => (
+    <SimpleModal
       isVisible={this.props.isVisible}
       onBackdropPress={this.props.onBackdropPress}
       onModalShow={this.unSubmitSignApply}>
       {this.renderSignView()}
-    </Modal>
-  );
-
-  private renderWebModal = () => (
-    // @ts-ignore: visible right name for prop if import from modal-react-native-web
-    <Modal
-      visible={this.props.isVisible}
-      onBackdropPress={this.props.onBackdropPress}
-      onModalShow={this.unSubmitSignApply}>
-      {this.renderSignView()}
-    </Modal>
+    </SimpleModal>
   );
 
   renderSignView = () => {
     const theme = this.state.colors[this.props.theme];
+    if (!this.props.isVisible) {
+      return null;
```

Comment:
the modal itself should be responsible for not rendering when it is not opened

Reply:
So this code is not necessary at the moment, the modal will not be rendered anyway